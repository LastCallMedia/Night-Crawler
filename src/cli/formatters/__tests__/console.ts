import format from '../console';
import { TestResultMap, TestResult } from '../../../testing/TestContext';

function r(obj: { [k: string]: TestResult }): TestResultMap {
  return new Map(Object.entries(obj));
}

describe('Console Formatter', function() {
  const each = new Map(
    Object.entries({
      ok: r({ ok: { pass: true } }),
      err: r({ err: { pass: false, message: 'failed' } })
    })
  );
  const all = r({
    time: { pass: true },
    errors: { pass: false, message: 'something failed' }
  });

  it('Should output results', function() {
    expect(format(each, all)).toMatchSnapshot();
  });

  it('Should display even when there are no results', function() {
    expect(format(new Map(), new Map())).toMatchSnapshot();
  });
});
