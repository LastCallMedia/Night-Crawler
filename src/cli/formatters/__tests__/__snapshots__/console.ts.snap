// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console Formatter Should display even when there are no results 1`] = `
"URL Results:
======
[32mNo results to display[39m

All Requests:
=======
[32mNo results to display[39m"
`;

exports[`Console Formatter Should output results 1`] = `
"URL Results:
======
╔═══╤═════╤════════╗
║   │ Url │ Errors ║
╟───┼─────┼────────╢
║ [32m✔[39m │ [32mok[39m  │        ║
╟───┼─────┼────────╢
║ [31m✖[39m │ [31merr[39m │ [31m- err[39m  ║
╚═══╧═════╧════════╝


All Requests:
=======
╔═══╤════════╗
║   │ Name   ║
╟───┼────────╢
║ [32m✔[39m │ [32mtime[39m   ║
╟───┼────────╢
║ [31m✖[39m │ [31merrors[39m ║
╚═══╧════════╝
"
`;
